src/NuGetForUnity/Editor/NugetAssetPostprocessor.cs

```diff
@@ -191,6 +221,44 @@ private static void ModifyImportSettingsOfRoslynAnalyzer([NotNull] PluginImporte
 
             AssetDatabase.SetLabels(plugin, new[] { RoslynAnalyzerLabel, ProcessedLabel });
 
+            // The nuget package can contain analyzers for multiple Roslyn versions.
+            // In that case, for the same package, the most recent version must be chosen out of those available for the current Unity version.
+            var assetRoslynVersion = GetRoslynVersionNumberFromAnalyzerPath(plugin.assetPath);
+            if (assetRoslynVersion != null)
+            {
+                var versionPrefixIndex = plugin.assetPath.IndexOf(AnalyzersRoslynVersionsFolderName, StringComparison.Ordinal);
+                var analyzersVersionsRoot = JoinPathComponents(plugin.assetPath.Substring(0, versionPrefixIndex), AnalyzersRoslynVersionsFolderName);
+                var analyzersFolders = AssetDatabase.GetSubFolders(analyzersVersionsRoot);
```

Comment:
Can we be sure `GetSubFolders` will work at this point since this is called from `OnPreprocessAsset` meaning maybe not all sub folders are yet imported?

Reply:
Ah this is a point of concern for me as well.

I installed the Nuget package on hand and tried it. (2021.3.16f, 2022.2.1f)

GetSubFolders seems to recognize folders before import is complete. 

Other methods I can think of are
- Directory.GetDirectories to query the file system directly.
- Rewrite the label in another Roslyn version.

However, I think GetSubFolders is simpler to implement. What do you think?

Reply:
If you test that it works correctly in these cases that is good enough for me:
1. Right after installing a package from Unity
2. Turn off Unity, delete the installed package but leave it in packages.json and start Unity and confirm it is automatically installed correctly on startup.
3. Turn off Unity, delete the installed package but leave it in packages.json and also delete the Library folder and then start Unity and confirm it is automatically installed correctly on startup.
4. If there are multiple installed packages repeat the previous step but delete them all while leaving them in packages.json.

Reply:
@igor84 I did the above test.

> Turn off Unity, delete the installed package but leave it in packages.json and start Unity

In this case, `AssetDatabase.GetSubFolders` does not seem to work, as you have pointed out.

So I rewrote the method to use `Directory.GetDirectories`. https://github.com/GlitchEnzo/NuGetForUnity/pull/616/commits/638c10a49e5bd8fc76849c1b272d25d0500caf98

With this commit, all steps 1-4 have passed.
Please check it out. Thanks.